src/store/rewards/saga.ts

```diff
@@ -29,6 +31,24 @@ export function* syncFetchRewards() {
   }
 }
 
+export function* checkNewRewardsLoaded() {
+  const zeroPreviousDay = yield select((state) => state.rewards.zeroPreviousDay);
+  const isFirstTimeLogin = yield select((state) => state.registration.isFirstTimeLogin);
+  if (!isFirstTimeLogin && zeroPreviousDay !== '0' && localStorage.getItem(lastViewedRewardsKey) !== zeroPreviousDay) {
+    yield put(setShowNewRewards(true));
+  }
+}
+
+export function* rewardsPopupClosed() {
+  // set last viewed rewards to the current rewards when the popup is closed
```

Comment:
What if they close the tooltip but not the rewards popup? We don't always show the rewards popup, that's only on registration, I think?

I think we should maybe set the last viewed at the same time we set the `showNewRewards`?

Reply:
We record that the user has viewed the rewards (by using `localStorage.last_viewed_rewards`) when it "views" the reward popup, and not the reward tooltip 😅 

Because you always have this status (shown below) that the user hasn't viewed the latest rewards
<img width="137" alt="image" src="https://github.com/zer0-os/zOS/assets/33264364/7cc21642-5715-4601-b1d5-4bba5dcf19fa">

After the user opens the popup & closes it, then we record that the user has viewed these (this is how it was originally implemented, it is just moved to the sagas here)

Reply:
ahhh, right... interesting... those seem like different things to me maybe we'll eventually split them. Eg: if I don't want to have to keep closing the tooltip and it's not obvious that opening the rewards would do that for me.

 For now, you're right, this is correct!